lib/node.js

```diff
@@ -170,4 +170,59 @@ Node.prototype.getSortedResults = function (prefix, results, opts, pqueue) {
   }
 };
 
+Node.prototype.delete = function (key, distinct) {
+  return this._deleteRecursive(key, distinct, 0);
+};
+
+Node.prototype._deleteRecursive = function (key, distinct, index) {
+  if (index === key.length || this.leaf) {
+    // If the index matches the key length, we are at the node to be deleted.
+    // Remove values associated with the key, where the `distinct` value also matches.
+    this.values = this.values.filter((item) => (item.key !== key && item.distinct !== distinct));
+
+    // If the node becomes a leaf node, update its status.
+    this.leaf = this.values.length > 0;
+
+    // If the node has no values or children, it can be pruned from the tree.
+    if (this.values.length === 0 && Object.keys(this.children).length === 0) {
+      return null; // Signal that this node should be removed.
+    }
+
+    // Node still has values or children, so no need to prune it.
+    return this;
+  }
+
+  const chr = key[index];
+  if (!this.children[chr]) {
```

Comment:
Is it expected that a `Trie` might look like (from the `should sort values by score` test):
```
Trie {
  root: Node {
    score: 4,
    children: {},
    values: [
      { key: 'aaa', value: 2, score: 2 },
      { key: 'aab', value: 3, score: 3 },
      { key: 'abb', value: 1, score: 1 },
      { key: 'aaa', value: 4, score: 4 }
    ],
    sorted: false,
    leaf: true
  },
  maxWidth: 500
}
```
My initial expectation was that there should be `children`, so I just want to make sure that this structure does not impact the highlighted lines.

I think in this case, the node would be detected as a `leaf` by the first condition in `_deleteRecursive`, so the deletion would work.

Reply:
I believe this is expected - From my testing so far, it seems like the tree rebalances once a node hits `maxWIdth` (in this case, 500), so the example (4 items) above does look like it is working as intended.